hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneServiceProvider.java

```diff
@@ -49,34 +49,24 @@
 
   @PostConstruct
   public void init() {
-    Collection<String> serviceIdList =
-        conf.getTrimmedStringCollection(OZONE_OM_SERVICE_IDS_KEY);
-    if (serviceIdList.size() == 0) {
-      // Non-HA cluster
+    try {
+      omserviceID = OmUtils.getOzoneManagerServiceId(conf);
+    } catch (IOException ex) {
+      throw new ConfigurationException(ex.getMessage(), ex);
```

Comment:
IIUC: if ozone-site.xml has more than one service id configured, s3g startup will fail as well based on OmUtils.getOzoneMangerServiceId.

Reply:
When it has more than one service id, it will fail only when internal service id is not configured.